alf/trainers/policy_trainer.py

```diff
@@ -0,0 +1,303 @@
+            use_tf_functions (bool): whether to use tf.function
+            checkpoint_interval (int): checkpoint every so many iterations
+            evaluate (bool): A bool to evaluate when training
+            eval_interval (int): evaluate every so many iteration
+            num_eval_episodes (int) : number of episodes for one evaluation
+            summary_interval (int): write summary every so many training steps
+            summaries_flush_secs (int): flush summary to disk every so many seconds
+            summary_max_queue (int): flush to disk every so mary summaries
+            debug_summaries (bool): A bool to gather debug summaries.
+            summarize_grads_and_vars (bool): If True, gradient and network variable
+                summaries will be written during training.
+        """
+        root_dir = os.path.expanduser(root_dir)
+        self._train_dir = os.path.join(root_dir, 'train')
+        self._eval_dir = os.path.join(root_dir, 'eval')
+
+        self._env = None
+        self._algorithm_ctor = algorithm_ctor
+        self._algorithm = None
+        self._driver = None
+
+        self._random_seed = random_seed
+        self._num_iterations = num_iterations
+        self._use_tf_functions = use_tf_functions
+
+        self._checkpoint_interval = checkpoint_interval
+        self._checkpointer = None
+
+        self._evaluate = evaluate
+        self._eval_interval = eval_interval
+        self._num_eval_episodes = num_eval_episodes
+        eval_metrics = None
+        eval_summary_writer = None
+        if self._evaluate:
+            eval_metrics = [
+                tf_metrics.AverageReturnMetric(
+                    buffer_size=self._num_eval_episodes),
+                tf_metrics.AverageEpisodeLengthMetric(
+                    buffer_size=self._num_eval_episodes)
+            ]
+            eval_summary_writer = tf.summary.create_file_writer(
+                self._eval_dir, flush_millis=summaries_flush_secs * 1000)
+        self._eval_env = None
+        self._eval_summary_writer = eval_summary_writer
+        self._eval_metrics = eval_metrics
+
+        self._summary_interval = summary_interval
+        self._summaries_flush_mills = summaries_flush_secs * 1000
+        self._summary_max_queue = summary_max_queue
+        self._debug_summaries = debug_summaries
+        self._summarize_grads_and_vars = summarize_grads_and_vars
+
+    def initialize(self):
+        """Initializes the Trainer."""
+
+        tf.random.set_seed(self._random_seed)
+        tf.config.experimental_run_functions_eagerly(
+            not self._use_tf_functions)
+        self._env = create_environment()
+        if self._evaluate:
+            self._eval_env = create_environment(num_parallel_environments=1)
+        self._algorithm = self._algorithm_ctor(
+            self._env, debug_summaries=self._debug_summaries)
+        self._driver = self.init_driver()
+
+    @abc.abstractmethod
+    def init_driver(self):
```

Comment:
Need more comments about what init_driver does because this is major interface for Trainer.
Args and Returns